### ISSUE TITLE
`do` blocks in `coffeeCompat` get treated as an argument to implicit application

### ISSUE THREAD

**Issue by STRd6** [DanielXMoore/Civet#660]
```civet
"civet coffeeCompat"

x
do ->
  y
```

Expected:

```typescript
x;
(function () {
  return y;
})();
```

Actual:

```typescript
x(function () {
  return y;
})();
```

This may just be a subset of https://github.com/DanielXMoore/Civet/issues/643

**Comment by edemaine** [comment]
Confirmed, it's an ASI issue.

**Comment by STRd6** [comment]
Closed as a dupe of #643
